src/layouts/AppContainer.tsx

```diff
@@ -81,48 +81,52 @@ class AppContainer extends React.Component<IProps, IState> {
   }
   public async componentWillMount() {
     // we initialize Arc
-    initializeArc().then(async () => {
-      // if Metamask is available, we wathc for any account changes
-      let metamask: any;
-      const currentAddress = await getCurrentAccountAddress();
-      if (currentAddress)  {
-        console.log(`using address from web3 connection: ${currentAddress}`);
-        this.props.cookies.set("currentAddress", currentAddress, { path: "/"});
-        this.props.setCurrentAccount(currentAddress);
-      } else {
-        const currentAddressFromCookie = this.props.cookies.get("currentAddress");
-        if (currentAddressFromCookie) {
-          console.log(`using address from cookie: ${currentAddressFromCookie}`);
-          this.props.setCurrentAccount(currentAddressFromCookie);
+    initializeArc()
```

Comment:
hmm, I changed how this works on the multiple-schemes branch. we now initializeArc way sooner in the app load process, so this will  be some merge conflicts...

Reply:
Ok. This is more about error handlign. And the automatic timeout so that  if something hangs in initalizeArc, we will give up after one sec...